script/hardhat/fork/govUpdateTest.ts

```diff
@@ -0,0 +1,326 @@
+import { impersonateAccount, loadFixture, mine, mineUpTo, setBalance } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { Comptroller, GovernorBravoDelegate, IAccessControlManager } from "../../../typechain";
+import { TimelockInterface } from "../../../typechain/contracts/Governance/GovernorAlpha2.sol";
+import { getCalldatas, setForkBlock } from "./utils";
+
+const ONE_HOUR = 800; // About 1 hour, 3 secs per block (20 * 60)
+const SIX_HOURS = 7200; // About 6 hours, 3 secs per block (20 * 60 * 6)
+const ONE_DAY = 28800; // About 24 hours, 3 secs per block (20 * 60 * 24)
+
+const ACL_MAINNET = "0x4788629ABc6cFCA10F9f969efdEAa1cF70c23555";
+const COMPTROLLER_PROXY_MAINNET = "0xfD36E2c2a6789Db23113685031d7F16329158384";
+const XVS_VAULT_MAINNET = "0x051100480289e704d20e9db4804837068f3f9204";
+const NORMAL_VIP_TIMELOCK = "0x939bD8d64c0A9583A7Dcea9933f7b21697ab6396";
+const GOVERNOR_PROXY_MAINNET = "0x2d56dC077072B53571b8252008C60e945108c75a";
+const PROPOSAL_TYPE_CONFIGS = [
+  // ProposalType.NORMAL
+  {
+    votingDelay: 1,
+    votingPeriod: ONE_DAY,
+    proposalThreshold: parseUnits("300000", 18),
+  },
+  // ProposalType.FASTTRACK
+  {
+    votingDelay: 1,
+    votingPeriod: ONE_DAY,
+    proposalThreshold: parseUnits("300000", 18),
+  },
+  // ProposalType.CRITICAL
+  {
+    votingDelay: 1,
+    votingPeriod: SIX_HOURS,
+    proposalThreshold: parseUnits("300000", 18),
+  },
+];
+
+const TIMELOCK_DELAYS_MAINNET = {
+  NORMAL: 172800,
+  FAST_TRACK: SIX_HOURS,
+  CRITICAL: ONE_HOUR,
+};
+
+const PROPOSAL_TYPES = {
+  NORMAL: 0,
+  FASTTRACK: 1,
+  CRITICAL: 2,
+};
+
+let comptrollerProxy: Comptroller;
+let accessControl: IAccessControlManager;
+let governorProxy: GovernorBravoDelegate;
+let proposer: SignerWithAddress;
+let supporter: SignerWithAddress;
+let governorAdmin: SignerWithAddress;
+let aclAdmin: SignerWithAddress;
+let timeLockFastTrack: TimelockInterface;
+let timeLockCritical: TimelockInterface;
+
+const initMainnetUser = async (user: string, balance: number) => {
+  await impersonateAccount(user);
+  await setBalance(user, balance);
+  return ethers.getSigner(user);
```

Comment:
if the goal is to just return the signer. We should we un-impersonate after. Otherwise the connected account remains the last impersonated